src/Compat/CompatController.php

```diff
@@ -420,6 +421,60 @@ public function sendExtraUpdates(array $updates)
         $this->getResponse()->setHeader('X-Icinga-Extra-Updates', join(',', $extraUpdates));
     }
 
+    /**
+     * Close the modal content and refresh the related view
+     *
+     * NOTE: When you use this with older Icinga Web versions (< 2.12), you will need to specify a valid redirect url,
+     * that will produce the same result as using the `__REFRESH__` redirect with the latest Icinga Web version.
+     *
+     *  This is supposed to be used in combination with a modal view and closes only the modal content,
+     * and refreshes the modal opener (regardless of whether it is col1 or col2).
+     *
+     * @param null|Url|string $url
+     * @param array<int|string, string> $extraUpdates See {@see static::sendExtraUpdates()} how to properly
+     *        provide extra updates
+     */
+    public function closeModalAndRefreshRelatedView($url = null, array $extraUpdates = []): void
+    {
+        $this->sendExtraUpdates($extraUpdates);
+
+        if (version_compare(Version::VERSION, '2.12.0', '<')) {
+            if (! $url) {
+                throw new InvalidArgumentException('No redirect url provided');
+            }
+
+            $this->redirectNow($url);
+        } else {
+            $this->redirectNow('__REFRESH__');
+        }
+    }
+
+    /**
+     * Close the modal content and refresh all the remaining views
+     *
+     * NOTE: When you use this with older Icinga Web versions (< 2.12), you will need to specify a valid redirect url,
+     * that will produce the same result as using the `__REFRESH__` redirect with the latest Icinga Web version.
+     *
+     * This is supposed to be used in combination with a modal view and closes only the modal content. It refreshes
+     * the modal opener (regardless of whether it is col1 or col2) and sends extra updates for col1.
```

Comment:
nope, it expects the opener to be col2, otherwise the hardcoded update for col1 wouldn't make any sense